File: virtual-world-cpp/VirtualWorld/Animals/Wolf.cpp
#include "Wolf.h"

Wolf::Wolf(int x, int y, World* world):
        Animal(x, y, 9, 5, 0, wolfCode, world) {}

bool Wolf::AttackPaired(Organism *attacker) {
    return false;
}

bool Wolf::TryToBreed(Organism *other) {
    if(dynamic_cast<Wolf*>(other) != nullptr)
        return true;
    return false;
}

std::string Wolf::GetName() {
    return "Wolf";
}

Organism *Wolf::Clone(int x, int y, World *world) {
    return new Wolf(x, y, world);
}

File: virtual-world-java/src/main/java/me/knasuta/virtualworld/simulation/animals/Wolf.java
package me.knasuta.virtualworld.simulation.animals;

import me.knasuta.virtualworld.simulation.*;
import me.knasuta.virtualworld.simulation.Point;

import java.awt.*;

public class Wolf extends Animal {
    public Wolf(Point location, World world){
        super(location, 9, 5, 0, world);
    }


    @Override
    public boolean AttackPaired(Organism other) {
        return false;
    }

    @Override
    public String getName() {
        return "Wolf";
    }

    @Override
    public Organism Clone(Point location, World world) {
        return OrganismFactory.getInstance().Create(OrganismType.WOLF, location, world);
    }

    @Override
    public Color getColor() {
        return Color.GRAY;
    }
}
